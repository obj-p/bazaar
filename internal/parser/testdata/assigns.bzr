package assigns

func Assigns() {
        var answer = 42
        var answer = 42.0
        var answer = "42"
        var answer int = 42
        var answer = { return 42 }

        answer += 0
        answer -= 0
        answer %= 128
        answer /= 1
        answer *= 1
}
