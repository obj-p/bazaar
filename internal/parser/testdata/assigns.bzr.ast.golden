&parser.Bazaar{
  Package: &parser.PackageDecl{
    Domain: []string{
      "assigns",
    },
  },
  Decls: []*parser.TopLevelDecl{
    {
      Function: &parser.FunctionDecl{
        Name: "Assigns",
        Block: []*parser.Stmt{
          {
            Var: &parser.VarDeclStmt{
              Dest: &parser.VariableExpr{
                Name: &"answer",
              },
              Source: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"42",
                    },
                  },
                },
              },
            },
          },
          {
            Var: &parser.VarDeclStmt{
              Dest: &parser.VariableExpr{
                Name: &"answer",
              },
              Source: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"42.0",
                    },
                  },
                },
              },
            },
          },
          {
            Var: &parser.VarDeclStmt{
              Dest: &parser.VariableExpr{
                Name: &"answer",
              },
              Source: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      String: &parser.String{
                        Fragments: []*parser.StringFragment{
                          {
                            Text: &"42",
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
          {
            Var: &parser.VarDeclStmt{
              Dest: &parser.VariableExpr{
                Name: &"answer",
              },
              Type: &parser.TypeDecl{
                Value: &"int",
              },
              Source: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"42",
                    },
                  },
                },
              },
            },
          },
          {
            Var: &parser.VarDeclStmt{
              Dest: &parser.VariableExpr{
                Name: &"answer",
              },
              Source: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Lambda: &parser.LambdaExpr{
                      Stmts: []*parser.Stmt{
                        {
                          Return: &parser.ReturnStmt{
                            Value: &parser.Expr{
                              Unary: &parser.UnaryExpr{
                                Primary: &parser.PrimaryExpr{
                                  Literal: &parser.Literal{
                                    Number: &"42",
                                  },
                                },
                              },
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          },
          {
            Assign: &parser.AssignStmt{
              Name: &"answer",
              Op: &token.Op(token.OpAddAssign),
              Value: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"0",
                    },
                  },
                },
              },
            },
          },
          {
            Assign: &parser.AssignStmt{
              Name: &"answer",
              Op: &token.Op(token.OpSubAssign),
              Value: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"0",
                    },
                  },
                },
              },
            },
          },
          {
            Assign: &parser.AssignStmt{
              Name: &"answer",
              Op: &token.Op(token.OpModAssign),
              Value: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"128",
                    },
                  },
                },
              },
            },
          },
          {
            Assign: &parser.AssignStmt{
              Name: &"answer",
              Op: &token.Op(token.OpDivAssign),
              Value: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"1",
                    },
                  },
                },
              },
            },
          },
          {
            Assign: &parser.AssignStmt{
              Name: &"answer",
              Op: &token.Op(token.OpMulAssign),
              Value: &parser.Expr{
                Unary: &parser.UnaryExpr{
                  Primary: &parser.PrimaryExpr{
                    Literal: &parser.Literal{
                      Number: &"1",
                    },
                  },
                },
              },
            },
          },
        },
      },
    },
  },
}
